### PR TITLE
Adding more trashbin scenarios

### PR DESCRIPTION
**Pull request by SergioBertolinSG**
Something is failing in the second scenario. The listed elements of the trashbin are less than expected and the file the case is looking for is not found.

### CODE REVIEW

**Comment by SergioBertolinSG** [comment]
The requests to ```index.php/apps/files_trashbin/ajax/list.php?dir=%2F```are not returning all the elements in the trashbin. It is having troubles with the files in subfolders.
Only happening when the files has the same name. @PVince81

File: tests/integration/features/trashbin-new-endpoint.feature [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		Given As an "admin"
+		And user "user0" exists
+		And user "user1" exists
+		And user "user0" created a folder "/shared"
+		And User "user0" moved file "/textfile0.txt" to "/shared/shared_file.txt"
+		And folder "/shared" of user "user0" is shared with user "user1"
+		When User "user1" deletes folder "/shared"
+		Then as "user1" the folder with original path "/shared" does not exist in trash
+
+	Scenario: deleting a file in a recieved folder doesn't moves it to trashbin
```
Comment (phil-davis) [inline tests/integration/features/trashbin-new-endpoint.feature]: There are a few "recieved" - here and further down.
s/recieved/received
Isn't English a great language ;)

**Comment by SergioBertolinSG** [comment]
Pending to be done: fixes for the two failing cases. Check if the whole suite pass after last changes.

**Comment by felixboehm** [comment]
If your suite contains tests, which point out bugs, is this a blocker for merging the tests?
Suggestion: Can you open an PR for each, containing the failing testcase, so that fixing and test come together in that PR?

**Comment by SergioBertolinSG** [comment]
Failing cases in jenkins:

```
trashbin-new-endpoint.deleting a file in a received folder it is moved to the trashbin
trashbin-new-endpoint.trashbin can store two files with same name but different origins
trashbin-old-endpoint.deleting a file in a received folder it is moved to the trashbin
trashbin-old-endpoint.trashbin can store two files with same name but different origins
trashbin-old-endpoint.Deleting a folder into external storage moves it to the trashbin
trashbin-old-endpoint.Deleting a file into external storage moves it to the trashbin and can be restored
```

Pending to implement the skip label to fix this cases in a different PR.

**Comment by PVince81** [comment]
are these cases reproducible manually ? AFAIK at least storing two files with different origins should still work.

**Comment by SergioBertolinSG** [comment]
Regarding "deleting a file in a received folder it is moved to the trashbin", yes I opened it here, the problem is a missing parameter called 'extraData': 

https://github.com/owncloud/core/issues/28121

Regarding "trashbin can store two files with same name but different origins", I cannot reproduce it manually, but in the tests it always fails.


trashbin-old-endpoint.Deleting a folder into external storage moves it to the trashbin

This one is a test problem, I can fix it.

trashbin-old-endpoint.Deleting a file into external storage moves it to the trashbin and can be restored 

This one has the same problem as the previous one, fixable in tests, but maybe the restoration part is affected also by the missing 'extraData' parameter bug.

**Comment by PVince81** [comment]
if manual reproduction doesn't work but automated test fails, then maybe the tests aren't doing exactly the same, or it's not using the same APIs.

**Comment by SergioBertolinSG** [comment]
Now the tests:

trashbin-new-endpoint.deleting a file in a received folder it is moved to the trashbin
trashbin-new-endpoint.trashbin can store two files with same name but different origins
trashbin-old-endpoint.deleting a file in a received folder it is moved to the trashbin
trashbin-old-endpoint.trashbin can store two files with same name but different origins

Are skipped. To be removed the "@skip" label to run when fixing the actual core issues.

**Comment by PVince81** [comment]
> trashbin-new-endpoint.deleting a file in a received folder it is moved to the trashbin

Is this reproducible manually ? Last time I tested this manually when working on this part all worked fine.

**Comment by PVince81** [comment]
Actually both the owner trashbin and recipient trashbin should receive the file.

**Comment by SergioBertolinSG** [comment]
@PVince81 the problem is the lack of extraData field. Which makes this step to fail:
Then as "user1" the file with original path "/shared/shared_file.txt" exists in trash

The file is in the trashbin, but if it is restored it will go to the root folder instead to the original path. https://github.com/owncloud/core/issues/28121

**Comment by PVince81** [comment]
ok

**Comment by PVince81** [comment]
Here is the fix for restoring into the wrong folder: https://github.com/owncloud/core/pull/28196

**Comment by PVince81** [comment]
See https://github.com/owncloud/core/blob/master/apps/files_trashbin/tests/TrashbinTest.php#L639 for more restore cases to cover

File: tests/integration/features/trashbin-new-endpoint.feature [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+			| /FOLDER/ |
+			| /PARENT/ |
+			| /PARENT/parent.txt |
+			| /textfile0.txt |
+			| /textfile1.txt |
+			| /textfile2.txt |
+			| /textfile3.txt |
+			| /textfile4.txt |
+
+	@skip
```
Comment (PVince81) [inline tests/integration/features/trashbin-new-endpoint.feature]: we still have skipped tests ?

Comment (SergioBertolinSG) [inline tests/integration/features/trashbin-new-endpoint.feature]: Yes, that scenario is not related with the extraData fix.

It doesn't fail manually, but in the test it always fail. I recommend to merge this and take a look when you want. The test doesn't seem wrong, looks like a hidden bug not reproducible using the ui.

**Comment by PVince81** [comment]
backport?

File: tests/integration/features/trashbin-new-endpoint.feature (hunk 2 above)
Comment (PVince81) [inline tests/integration/features/trashbin-new-endpoint.feature]: > hidden bug

I hate these

**Comment by SergioBertolinSG** [comment]
Ok, to stable10 and others? or just stable10?

**Comment by PVince81** [comment]
as far as possible I'd say to guarantee coverage on older versions too.
